csrc/python_frontend/python_bindings.cpp

```diff
@@ -2366,50 +2366,10 @@ void initNvFuserPythonBindings(PyObject* module) {
               opt_strides.value().size());
           strides.assign(
               opt_strides.value().begin(), opt_strides.value().end());
+        } else {
+          strides.resize(start_indices.size(), 1);
         }
-        NVF_CHECK(
```

Comment:
Can we keep these checks here?

There's a difference doing them here vs doing them in the `operator()`. i.e.

```
with FusionDefinition() as fd:
  ...
  fd.ops.slice([illegal_slice). # currently we throw error here.
  ...
```

Moving the checks into operator(), we only throw error when we construct fusion, i.e. when we exit the context manager.
Not a deal breaker, but would prefer to have semantic check at the place when the python API is called.

Reply:
alternatively, we can also do checks in `SliceOpRecord`:ctor.

Reply:
Good point! Done.